mods/tnt/init.lua

```diff
@@ -1,10 +1,10 @@
 tnt = {}
 -- Default to enabled in singleplayer and disabled in multiplayer
 local singleplayer = minetest.is_singleplayer()
-local setting = minetest.setting_getbool("enable_tnt")
-if (not singleplayer and setting ~= true) or
-		(singleplayer and setting == false) then
-	return
+local enable_tnt = minetest.setting_getbool("enable_tnt")
+if (not singleplayer and enable_tnt ~= true) or
+		(singleplayer and enable_tnt == false) then
+	enable_tnt = false
```

Comment:
What does this statement try to achieve?

If singleplayer is set and `enable_tnt` is **`false`**, then there is not point in setting it **`false`** again.

All this does is setting `enable_tnt` **`false`**, if and only if `singleplayer` is not set and `enable_tnt` is either **`false`** (in which case it is unnecessary to set it **`false`** again as well) or is **`nil`**.

So you could say it defaults to **`false`** if `singleplayer` is unset.
Which means this is logically equivalent to one of these:

```lua
local enable_tnt = minetest.setting_getbool("enable_tnt") or not not singleplayer
local enable_tnt = minetest.setting_getbool("enable_tnt") or singleplayer ~= nil
local enable_tnt = minetest.setting_getbool("enable_tnt") or not singleplayer == nil
```
But why? What the idea behind that?